File: 1-two-sum/two-sum.java


class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] arr = new int[2];

        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int requiredNum = target - num;

            if (map.containsKey(requiredNum)) {
                arr[0] = map.get(requiredNum);
                arr[1] = i;
                break;
            }
            map.put(num, i);
        }
        return arr;
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        n = len(nums)

        for i in range(n):
            digit1 = nums[i]
            rem = target - digit1
            for j in range(n):
                if rem == nums[j] and i != j:
                   result.append(i)
                   result.append(j)
                   return result
            i += 1
            
            





            
        